platforms/core-runtime/internal-instrumentation-processor/src/main/java/org/gradle/internal/instrumentation/extensions/property/PropertyUpgradeClassSourceGenerator.java

```diff
@@ -78,26 +85,67 @@ private static MethodSpec mapToMethodSpec(CallInterceptionRequest request, Consu
-                .addModifiers(Modifier.PUBLIC, Modifier.STATIC)
-                .addParameter(typeName(callable.getOwner().getType()), SELF_PARAMETER_NAME)
-                .addParameters(parameters)
-                .addCode(generateMethodBody(implementation, callable, implementationExtra))
-                .returns(typeName(callable.getReturnType().getType()))
-                .addAnnotations(getAnnotations(implementationExtra))
-                .build();
+            CallableInfo callable = request.getInterceptedCallable();
+
+            MethodSpec spec;
+            if (implementationExtra.getBridgedMethod() != null) {
+                spec = mapToBridgedMethod(implementation.getName(), implementationExtra, callable);
+            } else {
+                List<ParameterSpec> parameters = callable.getParameters().stream()
+                    .map(parameter -> ParameterSpec.builder(typeName(parameter.getParameterType()), parameter.getName()).build())
+                    .collect(Collectors.toList());
+                spec = MethodSpec.methodBuilder(implementation.getName())
+                    .addModifiers(Modifier.PUBLIC, Modifier.STATIC)
+                    .addParameter(typeName(callable.getOwner().getType()), SELF_PARAMETER_NAME)
+                    .addParameters(parameters)
+                    .addCode(generateMethodBody(implementation, callable, implementationExtra))
+                    .returns(typeName(callable.getReturnType().getType()))
+                    .addAnnotations(getAnnotations(implementationExtra))
+                    .build();
+            }
+
             onProcessedRequest.accept(request);
             return spec;
         } catch (Exception e) {
             onFailure.accept(new HasFailures.FailureInfo(request, e.getMessage()));
             throw e;
         }
+    }
+
+    private static MethodSpec mapToBridgedMethod(String methodName, PropertyUpgradeRequestExtra implementationExtra, CallableInfo callable) {
+        ExecutableElement bridgedMethod = implementationExtra.getBridgedMethod();
+        List<TypeVariableName> typeVariables = bridgedMethod.getTypeParameters().stream()
+            .map(element -> TypeVariableName.get((TypeVariable) element.asType()))
+            .collect(Collectors.toList());
+        List<ParameterSpec> parameters = bridgedMethod.getParameters().stream()
+            .map(ParameterSpec::get)
+            .collect(Collectors.toList());
+        List<TypeName> exceptions = bridgedMethod.getThrownTypes().stream()
+            .map(TypeName::get)
+            .collect(Collectors.toList());
+        String passedParameters = parameters.stream()
+            .map(parameterSpec -> parameterSpec.name)
+            .collect(Collectors.joining(", "));
+
+        CodeBlock.Builder bodyBuilder = CodeBlock.builder();
+        if (implementationExtra.getDeprecationSpec().isEnabled()) {
+            bodyBuilder.addStatement(getDeprecationCodeBlock(implementationExtra, callable));
```

Comment:
💭 For `$$BridgeFor$$` class methods we also generate deprecation

Reply:
I updated the PR description to include the fact that these methods are deprecated.